File: Data Structures & Algorithms/two-integer-sum/submission-2.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        //goal is to find the two indicies in an array thst would equal up to the target

        //brute force is to have two loops to check if the indicies equal the target 
        // return 0 if not

        // better algorithm would be to have a hash map and then add the numbers into that hash 
        // see wich ones add to the target and then return the indicies
        // similar algorithm to anagram one but with <int, int> instead

        unordered_map<int,int> count;

        for(int i = 0; i< nums.size(); i++){
            int c = target - nums[i];

            if(count.count(c)){
                return {count[c], i};
            }
                count[nums[i]] = i;
        }
      

        return {};
    }
};

File: Data Structures & Algorithms/two-integer-sum-ii/submission-2.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:

        #goal is similar to two sum
        #we want to find the two numbers
        #that equal to the target
        #the array is sorted and the indicies we need to return
        #will not be in 0-Base 

        #alg would be to use the two pointer method
        #we would first initialise left and right
        #as left and right moves we check if any of those number equal
        #the target
        #if not we return 

        left = 0
        right = len(numbers)-1

        while left < right:
            curr = numbers[left] + numbers[right]
            if curr == target:
                return [left+1, right+1]
            elif curr < target:
                left+=1
            else:
                right-=1

        